### PR TITLE
Add cuda metapackage recipes

### PR DESCRIPTION
**Pull request by adibbley**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/cuda/meta.yaml
Comment (bdice) [inline recipes/cuda/meta.yaml:18]: Alphabetize.
```suggestion
    - cuda-runtime {{ version }}
    - cuda-toolkit {{ version }}
```

Comment (bdice) [inline recipes/cuda/meta.yaml:18]: I'm confused by the structure here. `cuda` requires `cuda-runtime`, which requires `cuda-libraries`. But `cuda-toolkit` also requires `cuda-libraries`. Why do we need to list `cuda-runtime` here, and why do we need `cuda-runtime` at all?

**Review by bdice — COMMENTED** [review]
I have some questions like "why does `cuda-nvml-dev` not belong to `cuda-libraries-dev`?" and similarly for `gds-tools` not being a part of `cuda-command-line-tools` but I would guess that those questions are beyond the scope of conda-forge and are just part of the broader CUDA deployment decisions. This looks generally fine.

File: recipes/cuda-command-line-tools/meta.yaml
Comment (vyasr) [inline recipes/cuda-command-line-tools/meta.yaml:33]: I'm assuming that this is copy-pasted from NVIDIA docs? Otherwise I'd recommend aligning on the unhyphenated spelling of metapackage.

File: recipes/cuda-toolkit/README.md
Comment (vyasr) [inline recipes/cuda-toolkit/README.md:1]: Will conda-build follow symlinks (I would hope so)? If so we could use a single README and symlink it into all five recipes rather than copy-pasting it everywhere.

File: recipes/cuda-command-line-tools/meta.yaml
Comment (vyasr) [inline recipes/cuda-command-line-tools/meta.yaml:26]: Can we just omit the test section if it's not doing anything? Is the trivial installation of a metapackage worth testing?

Comment (vyasr) [reply recipes/cuda-command-line-tools/meta.yaml:26]: Same question applies to all the other recipes too.

File: recipes/cuda-command-line-tools/meta.yaml
Comment (vyasr) [inline recipes/cuda-command-line-tools/meta.yaml:1]: Is there a source of truth for what metapackages should exist and what goes in each? Are these just rough approximations of what we've previously done for e.g. apt or something?

**Review by vyasr — CHANGES_REQUESTED** [review]

File: recipes/cuda-toolkit/README.md
Comment (bdice) [reply recipes/cuda-toolkit/README.md:1]: These will become separate feedstocks so symlinks are not viable. This is a single PR but every directory in recipes/ becomes its own feedstock.

Comment (vyasr) [reply recipes/cuda-toolkit/README.md:1]: Oh missed that this was staging new ones. Never mind, you're right.

File: recipes/cuda-command-line-tools/meta.yaml
Comment (jakirkham) [reply recipes/cuda-command-line-tools/meta.yaml:26]: The linter complains if there are no tests

With metapackages, we use this approach

File: recipes/cuda/meta.yaml
Comment (jakirkham) [inline recipes/cuda/meta.yaml:37]: Let's go ahead and update this to include all the people we want here. Would suggest the following

```suggestion
extra:
  recipe-maintainers:
    - adibbley
    - bdice
    - jakirkham
    - leofang
    - vyasr
```

File: recipes/cuda-command-line-tools/meta.yaml
Comment (adibbley) [reply recipes/cuda-command-line-tools/meta.yaml:33]: Yes the summaries of all formats have used "Meta-package", although I agree that "metapackage" would be better.

File: recipes/cuda-command-line-tools/meta.yaml
Comment (adibbley) [reply recipes/cuda-command-line-tools/meta.yaml:1]: We don't ship any metadata specific to metapackages the way we do for components. It does exist internally, and was used to generate these.

**Review by bdice — APPROVED** [review]
I think everything looks good here, just need to apply the maintainers list suggestion. Are there other action items needed before merging?

File: recipes/cuda/meta.yaml
Comment (jakirkham) [reply recipes/cuda/meta.yaml:37]: cc @kmittman

**Comment by jakirkham** [comment]
Thanks all! 🙏

Let's follow up on anything else on the feedstock

**Pull request merged**
